### PR TITLE
Create user route

### PR DESCRIPTION
**Pull request by tomerpeceniski**

### CODE REVIEW

File: src/main/java/igym/services/UserService.java [hunk 1]
```diff
@@ -20,4 +22,12 @@ public List<User> findAll() {
         return repository.findAll();
     }
 
+    @Transactional
+    public User createUser(User user) {
+        if (repository.existsByName(user.getName()) == true) {
```
Comment (danssaul) [inline src/main/java/igym/services/UserService.java]: Unnecessary "==true"

File: src/test/java/igym/controllers/UserControllerTest.java [hunk 2]
```diff
@@ -1,50 +1,63 @@
...
+import igym.exceptions.DuplicateUserException;
 import igym.services.UserService;
+import jakarta.validation.Validator;
 
-@SpringBootTest
-@AutoConfigureMockMvc
-@ActiveProfiles("test")
+@WebMvcTest(UserController.class)
+@ExtendWith(MockitoExtension.class)
 public class UserControllerTest {
```
Comment (danssaul) [inline src/test/java/igym/controllers/UserControllerTest.java:39]: maybe it is missing the test case for empty string user creation like new User("")

File: src/test/java/igym/entities/UserTest.java [hunk 3]
```diff
@@ -2,32 +2,77 @@
...
+import igym.config.ValidationConfig;
+
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
+import jakarta.validation.*;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+
 public class UserTest {
```
Comment (danssaul) [inline src/test/java/igym/entities/UserTest.java:15]: and here missing the blank case

File: src/test/java/igym/entities/UserTest.java [hunk 4]
```diff
@@ -2,32 +2,77 @@
...
+    public void nullConstraintTest() {
+        User nullNameUser = new User();
+        Set<ConstraintViolation<User>> violations = validator.validate(nullNameUser);
+        assertTrue(violations.size() == 1);
+        ConstraintViolation<User> violation = violations.iterator().next();
+        assertEquals(NotBlank.class, violation.getConstraintDescriptor().getAnnotation().annotationType());
+    }
+
+    @Test
+    @DisplayName("Should return violation error due to blank name")
```
Comment (danssaul) [inline src/test/java/igym/entities/UserTest.java:50]: I think we both confused in this way, this should be an empty string and this " " blank, but let's define this and change the test into our pattern

**Review by danssaul — APPROVED** [review]
Just some details, check if your test cases are englobing all the cases and you can go ahead to merge. All code is much better than the last PR!
